buffer/src/main/java/io/netty5/buffer/api/internal/LeakDetection.java

```diff
@@ -61,6 +60,18 @@ public final class LeakDetection {
         leakDetectionEnabled = enabled;
     }
 
+    private static VarHandle getLeakDetectionEnabledUpdater() throws NoSuchFieldException, IllegalAccessException {
```

Comment:
This is a temporary workaround due to https://github.com/oracle/graal/issues/4744
`native-image` looks at the `findStaticVarHandle` method in a special way: if the arguments of this method are all constants, `native-image` will try and reflectively invoke this method during the image build and embed the result directly in the code.
However, in the case of `findStaticVarHandle`, this will trigger the initialization of the target class during the image build: while invoking this method, we will eventually reach [this snippet](https://github.com/graalvm/labs-openjdk-17/blob/master/src/java.base/share/classes/java/lang/invoke/VarHandles.java#L119):
```
if (UNSAFE.shouldBeInitialized(refc))
    UNSAFE.ensureClassInitialized(refc);
```
This will be fixed in GraalVM 22.3 - but until then, without this fix, `LeakDetection` would be initialized during the image build.
This workaround doesn't let `native-image` see the `fieldName` parameter (`leakDetectionEnabled`) as a constant, causing `native-image` to skip folding this call